framework/LinePlot/LinePlot.swift

```diff
@@ -115,7 +143,7 @@ extension LineGraph{
   }
 
   // functions implementing plotting logic
-  func calcLabelLocations(renderer renderer : Renderer){
+  func calcLabelLocations(renderer renderer : inout Renderer){
```

Comment:
Is there a reason for the switch back to `inout` on the Renderer interfaces? Is this pulling from older code where that had been the case? I think it still makes sense to treat the Renderers as reference types, and the interface was nicely cleaned up through the removal of the `inout`, etc. in one of the more recent commits.

Reply:
I wanted to pass the renderer as reference to the plot so that, when I am plot ring subplots, I can pass the same renderer object to all the plot objects and in the end just save the output.

Reply:
Isn't passing by reference better?

Reply:
If your Renderer is a reference type (a class), it should be natively passed by reference, even through to subplots. This may require slightly changing the definition of Renderer to 

```
public protocol Renderer: AnyObject {
```

so that it's understood that Renderer will always be a reference type. Thinking about it more, having Renderer as a value type could lead to problems with state changes among sub-operations, so I think we'll always want it to be a value type.

Reply:
Ok. I'll make the changes.

Reply:
Other than the inout issue is this ready to merge? I'll think I can take a look at the AGG buffer issue later, because the SubPlot logic is independent, therefore it'll stay the same.

I have removed inout, and am including the changes in the PR for functions, which I should be able to open by tonight.

Reply:
If you'll make the value->reference type changes later, or in a separate pull request, I'm fine with pulling this in.